tests/run_tests.sh

```diff
@@ -9,5 +9,5 @@ fi
 for t in $(realpath $(dirname $0)/test*.sh); do
     desc=$(sed -n 's/^# *//; 3p' $t)
     echo -n "$(basename $t) $desc..."
-    $t || exit 1
+    $t || echo FAIL
```

Comment:
maybe instead we should make err_exit always echo FAIL, and be sure that we use it everywhere.  That way, when you run the tests by themselves instead of via run_tests.sh, it will also show FAIL.

Reply:
Oh, we should save the return value though and exit at the end after running all of the tests - I missed the echo/exit change which is worth keeping.